pkg/controllers/image-loader-controller.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright © 2021 Microshift Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	lb "github.com/containers/common/libimage"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage"
+	"github.com/openshift/microshift/pkg/config"
+	"k8s.io/klog/v2"
+)
+
+type OCIImageLoader struct {
+	options openshift_apiserver.OpenShiftAPIServer
+	cfg     *config.MicroshiftConfig
+}
+
+func NewOCIImageLoader(cfg *config.MicroshiftConfig) *OCIImageLoader {
+	s := &OCIImageLoader{}
+	s.load(cfg)
+	return s
+}
+
+func (s *OCIImageLoader) Name() string { return "oci-image-loader" }
+func (s *OCIImageLoader) Dependencies() []string {
+	return []string{""}
+}
+func (s *OCIImageLoader) load(cfg *config.MicroshiftConfig) error {
+
+	runtime, cleanup := getRuntime()
+	defer cleanup()
+
+	ctx := context.Background()
+	loadOptions := &LoadOptions{}
+	loadOptions.Writer = os.Stdout
+
+	for _, listOfImages := range []struct {
+		input       string
+		expectError bool
+		numImages   int
+		names       []string
+	}
+
+	// OCI ARCHIVE 
+
+	{"/var/lib/microshift/tars/cli.tar", false, 1, []string{ "quay.io/microshift/cli" }},
+	{"/var/lib/microshift/tars/coredns.tar", false, 1, []string{"quay.io/microshift/coredns"}},
+	{"/var/lib/microshift/tars/haproxy_router.tar", false, 1, []string{"quay.io/microshift/haproxy-router"}},
+	{"/var/lib/microshift/tars/kube_flannel.tar", false, 1, []string{"quay.io/microshift/flannel"}},
+	{"/var/lib/microshift/tars/kube_flannel_cni.tar", false, 1, []string{"quay.io/microshift/flannel-cni"}},
+	{"/var/lib/microshift/tars/kube_rbac_proxy.tar", false, 1, []string{ "quay.io/microshift/kube-rbac-proxy"}},
+	{"/var/lib/microshift/tars/kubevirt_hostpath_provisioner.tar", false, 1, []string{"quay.io/microshift/hostpath-provisioner"}},
+	{"/var/lib/microshift/tars/pause.tar", false, 1, []string{"quay.io/microshift/pause:" + Base}},
+	{"/var/lib/microshift/tars/service_ca_operator.tar", false, 1, []string{"quay.io/microshift/service-ca-operator"}},
```

Comment:
Syntax error here - the slice of struct literals should be wrapped in an outer set of curly braces, e.g.

```
var _ = []struct{
    foo string
    bar string
}{ <---
{...},
{...},
{...},
} <---